### PR TITLE
Make DNS validators return error strings

### PR DESCRIPTION
**Pull request by thockin**
Part of the larger validation PR, broken out for easier review and merge.  Builds on previous PRs in the series.

### CODE REVIEW

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func (pm *VolumePluginMgr) InitPlugins(plugins []VolumePlugin, host VolumeHost)
 	allErrs := []error{}
 	for _, plugin := range plugins {
 		name := plugin.Name()
-		if !validation.IsQualifiedName(name) {
+		if len(validation.IsQualifiedName(name)) != 0 {
```
Comment (lavalamp) [inline pkg/volume/plugins.go]: Don't throw away the return value?

File: pkg/api/validation/name.go [hunk 2]
```diff
@@ -28,36 +28,36 @@ var NameMayNotBe = []string{".", ".."}
 var NameMayNotContain = []string{"/", "%"}
 
 // IsValidPathSegmentName validates the name can be safely encoded as a path segment
-func IsValidPathSegmentName(name string) (bool, string) {
+func IsValidPathSegmentName(name string) []string {
```
Comment (lavalamp) [inline pkg/api/validation/name.go]: It is a tad odd to return []string instead of []error, but presumably you considered the tradeoffs.

**Comment by thockin** [comment]
rebased and pushed

File: pkg/api/validation/validation.go [hunk 3]
```diff
@@ -163,9 +163,11 @@ func ValidateOwnerReferences(ownerReferences []api.OwnerReference, fldPath *fiel
 }
 
 // ValidateNameFunc validates that the provided name is valid for a given resource type.
-// Not all resources have the same validation rules for names. Prefix is true if the
-// name will have a value appended to it.
-type ValidateNameFunc func(name string, prefix bool) (bool, string)
+// Not all resources have the same validation rules for names. Prefix is true
+// if the name will have a value appended to it.  If the names is not valid,
```
Comment (krousey) [inline pkg/api/validation/validation.go]: "If the name"

**Comment by krousey** [comment]
LGTM with one minor English nit. Fix and add label.

**Comment by krousey** [comment]
Well... and fix 

```
cluster/addons/dns/kube2sky/kube2sky.go:211: invalid operation: exists && "k8s.io/kubernetes/pkg/util/validation".IsDNS1123Label(hostRecord.HostName) (mismatched types bool and []string)
```

too.

**Comment by thockin** [comment]
nits fixed and repushed

**Comment by thockin** [comment]
@k8s-bot test this: github issue #IGNORE

**Pull request merged**